odl/tomo/backends/astra_cpu.py

```diff
@@ -104,26 +105,47 @@ def astra_cpu_forward_projector(vol_data, geometry, proj_space, out=None):
     proj_id = astra_projector(vol_interp, vol_geom, proj_geom, ndim,
                               impl='cpu')
 
-    # Create ASTRA data structures
-    vol_id = astra_data(vol_geom, datatype='volume', data=vol_data,
-                        allow_copy=True)
+    if vol_data.space.is_rn:
```

Comment:
Hm, right. Check out the new version. However, I'm not sure the memory overhead is really less. The `writable_array` function will make copies since real and imag parts are not contiguous as needed by Astra.